assets/js/base/context/cart-checkout/checkout/processor/index.js

```diff
@@ -262,6 +206,94 @@ const CheckoutProcessor = () => {
 		orderNotes,
 		shouldCreateAccount,
 	] );
+
+	// Subscribe validation.
+	useEffect( () => {
+		let unsubscribeProcessing;
+		if ( ! expressPaymentMethodActive ) {
+			unsubscribeProcessing = onCheckoutBeforeProcessing(
+				checkValidation,
+				0
+			);
+		}
+		return () => {
+			if ( ! expressPaymentMethodActive ) {
+				unsubscribeProcessing();
+			}
+		};
+	}, [
+		onCheckoutBeforeProcessing,
+		checkValidation,
+		expressPaymentMethodActive,
+	] );
+
+	// If express payment method is active then suppress notices.
+	useEffect( () => {
+		setIsSuppressed( expressPaymentMethodActive );
+	}, [ expressPaymentMethodActive, setIsSuppressed ] );
+
+	// If an error is about to occur, dispatch it.
+	useEffect( () => {
+		if (
+			checkoutWillHaveError !== checkoutHasError &&
+			( checkoutIsProcessing || checkoutIsBeforeProcessing ) &&
+			! expressPaymentMethodActive
+		) {
+			dispatchActions.setHasError( checkoutWillHaveError );
+		}
+	}, [
+		checkoutWillHaveError,
+		checkoutHasError,
+		checkoutIsProcessing,
+		checkoutIsBeforeProcessing,
+		expressPaymentMethodActive,
+		dispatchActions,
+	] );
+
+	// Keep refs up to date.
+	useEffect( () => {
+		currentBillingData.current = billingData;
+		currentShippingAddress.current = shippingAddress;
+		currentRedirectUrl.current = redirectUrl;
+	}, [ billingData, shippingAddress, redirectUrl ] );
+
+	// When the billing or shipping address changes we need to push the changes to the server to
```

Comment:
This is the code addition (the other changes were just grouping by hook usage).

This effect monitors for address changes to both billing and shipping so they are updated together in a single request.